src/sentry/search/snuba/backend.py

```diff
@@ -401,11 +379,26 @@ def _query(self, projects, retention_window_start, group_queryset, tags, environ
                     result_group_ids.add(group_id)
                     result_groups.append((group_id, group_score))
 
+                if count_hits:
+                    if not more_results:
+                        # We know we have got all possible groups from snuba and filtered
+                        # them all down, so we have all hits.
+                        # TODO this probably doesn't work because we could be on page N
```

Comment:
This is the main problem in this PR. Basically the same thing as the "is there a previous page" problem that brett encountered before, but this time it makes the whole hit count bogus.

There might be a way to fall back to using the hits estimate in this case, as that never includes the > cursor_value condition in this case.

Reply:
There will also be a problem where the sample (and thus the estimated results) changes with every page as we paginate. We could fix by either making the sampling deterministically random (based on hash of query params and event_id or something) or rounding the hits number to (`> 1000`, `> 10000`, etc) although that will still have discontinuities if the estimate changes from 1001 to 999